### ISSUE TITLE
Some crates do not have 6 digits in timestamps

### ISSUE THREAD

**Issue by lodimas** [museun/cargo-whatfeatures#75]
Some crates, like polars, sometimes express timestamps with less than 6 digits in subsecond. I think format 

    const FORMAT: &[FormatItem<'static>] = time::macros::format_description!(
        "[year]-[month]-[day]T\
        [hour]:[minute]:[second].[subsecond digits:6]\
        [offset_hour sign:mandatory]:[offset_minute]"
    );

in client.rs could be adapted to this.

Incidentally, it would have been easier to see what was going on if line 91 of lib.rs exposed somehow the underlying error instead of discarding it with _. Assuming that errors can come only from non-existing or yanked crates is misleading and perhaps too much to expect.

**Comment by museun** [comment]
That's interesting, the timestamps are created by cargo for the crates.io API. I may have missed an api change for this, I'll investigate making this less fragile. 

Is this visible when doing `--list`? What about `--verbose`?

I've noticed some crates cause some internal errors more frequently recently.

More examples could be helpful to see if I need to handle crates.io internal changes

**Comment by museun** [comment]
On a side note, since I wrote this tool, more parts of cargo's internal API has been stabilized. And since then the registries API has become a thing. Currently I don't support registries outside of crates.io. 

Also the sparse index became stabilized and I should investigate how this can use that.

Error reporting could definitely be improved. I think I should improve it first and foremost.

**Comment by museun** [comment]
If I change it to:
```rust
    const FORMAT: &[FormatItem<'static>] = time::macros::format_description!(
        "[year]-[month]-[day]T\
        [hour]:[minute]:[second].[subsecond digits:1+]\
        [offset_hour sign:mandatory]:[offset_minute]"
    );
```

It should be backward and forward compatible. It just needs to parse any number of subsecond digits. 
With that change, it successfully lists all of the versions for polars.

As for the 'vague error' I don't know what could be done. 
This is where the parsing error is 'handled'

https://github.com/museun/cargo-whatfeatures/blob/e0c0c5640eee7a5b576b956ad527adbd76cb884f/src/client.rs#L52-L62

If I print the context, we'd get some serde error, then the context error of 'cannot list versions'
and then 'cannot find crate, maybe it was yanked'

The user shouldn't ever encounter a serde error. If one is to be produced, I think it'd be better for it print out that an internal error was encountered and that a github issue should be opened.

**Comment by museun** [comment]
Actually, a better fix is to use time's rfc3339 parsing. After some investigation, internally uses chrono for their time handling. And about 7 months ago chrono fixed their date time serialization to actually use a well formed rfc3339 instead of a bespoke one.

I implemented it in: #76

But, that only solves the problem and not prevents future problems.

I'll keep this open until I decide how internal errors -> bug reporting should be handled.

I published v0.9.11 which implements the timestamp parsing fix.

https://crates.io/crates/cargo-whatfeatures/0.9.11
